### ISSUE TITLE
Problems rendering trivial React code

### ISSUE THREAD

**Issue by bebraw** [interactivethings/catalog#149]
I just tried a simple specimen declaration like this:

```
\```react
class Demo extends React.Component {
  render() {
    console.log('rendering demo');

    return <div>foo</div>;
  }
}

<Demo />
\```
```

The code renders but `foo` doesn't. If I remove the `class Demo` declaration, it fails to work as expected, but if the code is there, nothing renders.

I might be missing something fundamental here. Note that I've imported React through `imports` so it definitely should find it. Just not rendering for some reason.

I noticed the official examples always rely on a component that's handed to the specimen through an import. But in my case I need to show the code that leads to the result as well so people can copy and paste. So far I've included the code separately but that's painful to maintain.

**Comment by herrstucki** [comment]
This is a tricky one to solve. What currently happens:

1. The code is transformed with babel-standalone
2. The first occurrence of `React.createElement` in the transformed code is prepended with `return` (in order for the next step to work)
3. A function is created with this code as body, Catalog imports are passed as function arguments and the function is executed.

Step 2 is the culprit because it turns the `return <div>foo</div>` into `return; return React.createElement('div', null, 'foo')` while not touching the `<Demo />` part.

The reason for this simple heuristic is that the React Specimen was never really intended to contain arbitrary code – just React elements.

I'm not sure if creating components on the fly in React Specimens is really necessary … you can get pretty far without it by using `state` and `setState` in the Specimen (see https://interactivethings.github.io/catalog/#/specimens/react). But maybe I'm missing some use case which is only possible by creating a class specifically for the documentation …

I can imagine two ways forward from here:

1. Make the transform more robust. I don't really know _how_ yet; it might involve using the AST instead just the generated code from the transpiler.
2. Enforce only React elements and displaying an error when other code shows up outside of one.

**Comment by bebraw** [comment]
Would it help if you would provide `render(<Foo />)` kind of API? If you detect `render`, you would use a modified execution path. Quick idea.

You can see the problem on a page like http://reactabular.js.org/#/examples/tree-view . I have to duplicate the React code. Some kind of import might be another way to handle it.

**Comment by herrstucki** [comment]
I improved the JSX transform a bit. It now should only pick up the first JSX element/call to React.createElement on the _top level_ (so not inside functions or class `render()` methods). I hope this fixes your issue!

**Comment by herrstucki** [comment]
Published in `v2.3.0-beta.2`, `v2.3.0` coming soon 😄

**Comment by bebraw** [comment]
Seems to work. Thanks a lot!
